lib/_http_incoming.js

```diff
@@ -144,6 +146,22 @@ ObjectDefineProperty(IncomingMessage.prototype, 'trailers', {
   }
 });
 
+ObjectDefineProperty(IncomingMessage.prototype, 'signal', {
+  get: function() {
+    if (!this[kAbortController]) {
+      this[kAbortController] = new AbortController();
+      if (this.aborted) {
+        this[kAbortController].abort();
```

Comment:
The abort here should maybe be fired in a `process.nextTick()`, otherwise there won't be any opportunity for any listeners to attach before the `abort()` is called. That could be fine, but it's not clear. What do you think @mcollina

Reply:
I agree

Reply:
I understand the rational for this change, but if I understand correctly this would mean that whoever is using the signal is using it incorrectly, as they should always check if it's `aborted` anyway, before adding a listener.